### PR TITLE
Support OSB API response schemas

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -539,8 +539,8 @@
     "v2",
     "v2/fake"
   ]
-  revision = "39dbb4e91dfc6aad1f293e73fba18cc5f75eceb3"
-  version = "0.0.6"
+  revision = "c57d7cdbb99a641570f006b9a063c7376306a3e9"
+  version = "0.0.8"
```
Comment (carolynvs) [inline Gopkg.lock]: Did you intend to upgrade to 0.0.8 of the osb client? The manifest and your comment said "0.0.7". Just making sure that we are not relying on changes is 0.0.8 unintentionally.

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 2]
```diff
@@ -419,8 +419,19 @@ type CommonServicePlanSpec struct {
...
+
+	// Currently, this field is ALPHA: it may change or disappear at any time
+	// and its data will not be migrated.when a bind operation stored in the Secret when binding to a ServiceInstance on this plan.
+	//
+	// ServiceBindingCreateResponseSchema is the schema for the response that
+	// will be returned by the broker when binding to a ServiceInstance on this plan.
+	// The schema also contains the sub-schema for the credentials part of the
+	// broker's response, which allows clients to see what the credentials
+	// will look like even before the binding operation is performed.
+	// TODO: introduce new struct with both Parameter and Response schemas?
```
Comment (carolynvs) [inline pkg/apis/servicecatalog/v1beta1/types.go]: Is this todo still valid?

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -1128,7 +1131,18 @@ func TestCatalogConversionWithParameterSchemas(t *testing.T) {
...
+			t.Fatalf("Unexpected value of ServiceBindingCreateParameterSchema; expected %v, got %v", e, a)
+		}
+	}
+
+	if plan.Spec.ServiceBindingCreateResponseSchema == nil {
+		t.Fatalf("Expected plan.ServiceBindingCreateResponseSchema to be set, but was nil")
+	}
+	m = make(map[string]string)
+	if err := json.Unmarshal(plan.Spec.ServiceBindingCreateResponseSchema.Raw, &m); err == nil {
+		//assert.Equal(t, "qax", "qux", "Unexpected value of ServiceBindingCreateResponseSchema")
```
Comment (carolynvs) [inline pkg/controller/controller_test.go]: Can this be removed now?

**Comment by pmorie** [comment]
CI Failure:

```
contrib/pkg/brokerapi/schemas.go:39:1: comment on exported type InputParametersSchema should be of the form "InputParametersSchema ..." (with optional leading article)
```

File: Gopkg.lock (hunk 1 above)
Comment (luksa) [reply Gopkg.lock]: Ah, I didn't notice dep pulled in a later version. I'll change the dependency constraint to "=0.0.7".

File: pkg/apis/servicecatalog/v1beta1/types.go (hunk 2 above)
Comment (luksa) [reply pkg/apis/servicecatalog/v1beta1/types.go]: I'll remove it for now, but IMHO we should reorganize all the schema-related fields before moving them out of Alpha (they are currently all directly under `ServicePlanSpec`).

File: pkg/controller/controller_test.go (hunk 3 above)
Comment (luksa) [reply pkg/controller/controller_test.go]: Huh, yeah. Don't know how I missed that when reviewing. Thanks

File: pkg/apis/servicecatalog/types.go [hunk 4]
```diff
@@ -384,6 +384,16 @@ type CommonServicePlanSpec struct {
...
+
+	// Currently, this field is ALPHA: it may change or disappear at any time
+	// and its data will not be migrated.when a bind operation stored in the Secret when binding to a ServiceInstance on this plan.
+	//
+	// ServiceBindingCreateResponseSchema is the schema for the response that
+	// will be returned by the broker when binding to a ServiceInstance on this plan.
+	// The schema also contains the sub-schema for the credentials part of the
+	// broker's response, which allows clients to see what the credentials
+	// will look like even before the binding operation is performed.
+	ServiceBindingCreateResponseSchema *runtime.RawExtension
```
Comment (pmorie) [inline pkg/apis/servicecatalog/types.go]: We should have a feature gate for this, since it's alpha.

Comment (luksa) [reply pkg/apis/servicecatalog/types.go]: Done.

**Pull request merged**